cmd/dagger/exec_unix.go

```diff
@@ -0,0 +1,26 @@
+//go:build unix
+// +build unix
+
+package main
+
+import (
+	"os/exec"
+	"syscall"
+)
+
+func ensureChildProcessesAreKilled(cmd *exec.Cmd) {
+	if cmd.SysProcAttr == nil {
+		cmd.SysProcAttr = &syscall.SysProcAttr{}
+	}
+
+	// cmd.SysProcAttr.Pdeathsig = syscall.SIGKILL
+	cmd.SysProcAttr.Setpgid = true
+
+	cmd.Cancel = func() error {
+		pgid, err := syscall.Getpgid(cmd.Process.Pid)
+		if err != nil {
+			return err
+		}
+		return syscall.Kill(-pgid, syscall.SIGKILL)
```

Comment:
iiuc this is what we'll send if `dagger run` gets `SIGINT`, right? If so, sending `SIGKILL` right away seems a little aggressive

Reply:
Also fires if you press `q`.

I can be talked out of it, but this is intentional. Telling a command to exit and then having it hang is incredibly frustrating, especially when it's a program that completely takes over your terminal, and especially when it happens constantly as part of a tight feedback loop. IMO everything should just try to be durable to a `kill -9` instead, since you often have to deal with nasty exit states anyway.

I figure we can get away with especially in this context where just about everything should be containerized and easy to re-run with minimal "pollution" from abrupt exits. But I'll watch for feedback here.

Reply:
I think the problem is that this is used for `dagger run`, so the command the user is executing may actually require not getting sigkilled if it needs to do some cleanup. I think it's probably at least a good idea to make this configurable at some point

Reply:
I'll can switch it to something softer and see if anything starts hanging for me, maybe I'm chasing ghosts.

I get that people like to do all sorts cleanup when interrupted, I just have an unusually extremist position that they shouldn't do that. :sweat_smile: It's awful UX (slowness, timeouts, stuck states, bricked terminal) and doesn't prevent you from having to deal with leaked resources anyway. (What if the dev was impatient that day and just spammed Ctrl+C? What if your laptop battery died? What if the application panicked and it wasn't trapped? What if your pipelines call `os.Exit(1)`? etc.)

In short I've never met a cleanup handler I liked. I'd rather just have a cleanup script.

Reply:
Changed it to SIGTERM + 10 second timeout (`--cleanup-timeout`) => SIGKILL but let me know if you've got a different preference for the default!